### ISSUE TITLE
ffmpeg and several other errors mostly with direct stream.

### ISSUE THREAD

**Issue by Liquidbings** [DEFENDORe/pseudotv#33]
[mpegts @ 0x563facec66e0] Timestamps are unset in a packet for stream 0. This is deprecated and will stop working in the future. Fix your code to set the timestamps properly
[mpegts @ 0x563facec66e0] first pts value must be set
av_interleaved_write_frame(): Invalid data found when processing input
frame= 1 fps=0.0 q=-1.0 Lsize= 18kB time=-00:00:00.52 bitrate=N/A speed=N/A
video:14kB audio:13kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Conversion failed!

FFMPEG ERROR {

code: 1,
cmd: '/usr/bin/ffmpeg -threads 20 -ss 1720.252 -t 4669.172 -re -i http://192.168.1.25:32400/library/parts/9553/1562451807/file.avi?X-Plex-Token=qmtjiNjuBiCVNdAKfRro -map VIDEOSTREAM -map 0:1 -c:v copy -c:a copy -metadata service_provider="PseudoTV" -metadata service_name="RiffTrax" -f mpegts -output_ts_offset 0 -muxdelay 0 -muxpreload 0 pipe:1'
}
_http_outgoing.js:526
throw new ERR_HTTP_HEADERS_SENT('set');
^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client

at ServerResponse.setHeader (_http_outgoing.js:526:11)
at ServerResponse.header (/home/node/app/node_modules/express/lib/response.js:771:10)
at ServerResponse.contentType (/home/node/app/node_modules/express/lib/response.js:599:15)
at ServerResponse.send (/home/node/app/node_modules/express/lib/response.js:145:14)
at FFMPEG.<anonymous> (/home/node/app/src/video.js:75:29)
at FFMPEG.emit (events.js:310:20)
at ChildProcess.<anonymous> (/home/node/app/src/ffmpeg.js:208:22)
at ChildProcess.emit (events.js:310:20)
at maybeClose (internal/child_process.js:1021:16)
at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5) {
code: 'ERR_HTTP_HEADERS_SENT'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pseudotv@1.0.0 start: `node index.js`
npm ERR! Exit status 1

npm ERR! Failed at the pseudotv@1.0.0 start script.

npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR! /root/.npm/_logs/2020-05-20T22_32_46_547Z-debug.log

> pseudotv@1.0.0 start /home/node/app
> node index.js

This happens often when I try and switch channels and the docker crashes and shuts off. 

I also can't seem to get Nvenc working. I have it working in plex for HW transcoding but using the same variables in docker config and turning it on in pseudotv just causes it to fail. 

Other than that absolutely loving the work here and thank everyone involved very kindly. I'll glad to give any more info or assistance where I can to help. I was wondering if you guys mind if I share the link for this over on the PseudoTv/Pseudotv Live page on the kodi forums as there a bunch of people over there waiting for new development on something like this and I believe they would be as excited as I or more and would likely want to assist.

**Comment by powdermilkman** [comment]
Of course you can link to this project. As far as Nvenc goes, are you using the nvidia unraid template?

The ffmpeg error is known and something we plan to design better with the rewrite. For now direct play works best if you have a channel consisting of a single codec/container that you know is working.

**Comment by Liquidbings** [comment]
Thank you kindly for the quick response. Appreciate the feedback. I've gotten a few other errors I've forgotten to document. But will be avid on it for now. I sent the link over the the great folks over on the Kodi site and hopefully a few will pop over here and give some hands in helping out 👍👌💯❤️

For the Nvenc correct I do use the Nvidia unraid template and have it working correctly for Plex and emby. 

I tried the direct play method as I was having trouble getting the transcoding working well. I had lots of stuttering, pausing and crashing. I'm still learning about much of the finer details of how alot of this works so wasn't sure on a bunch of the settings. 

Definitely learned tons in the last year and loving every moment of it. Thanks again for all of everyone's hard work on this amazing project. I've been a fan of the Kodi pseudotv for years and this project seems like the next best step for this project. Much love and thanks

**Comment by powdermilkman** [comment]
When transcoding try using the hevc_nvenc video encoder

**Comment by Liquidbings** [comment]
> When transcoding try using the hevc_nvenc video encoder

When I try it won't play any videos. Let me set it up again and see what the log gives us

**Comment by powdermilkman** [comment]
What video card are you using?

**Comment by Liquidbings** [comment]
1050ti

**Comment by powdermilkman** [comment]
`[hevc_nvenc @ 0x557ee048b280] Cannot load libcuda.so.1`
Are you sure your docker container has the nvidia runtime loaded? The parameter is --runtime=nvidia

**Comment by Liquidbings** [comment]
durr... Should have figured I missed something so simple... 

Working again. not as quick to swap channels as direct play but smooth unlike lib Appreciated. Keep up the amazing work bud

**Comment by powdermilkman** [comment]
If you didn't notice there is a pre-made template for [pseudotv on unraid](https://github.com/DEFENDORe/pseudotv#unraid-install)
Glad it's working for you, things should get a lot better with the rewrite.

**Comment by Liquidbings** [comment]
Yeah That was how I got it working. I tried the old way back when you were on gitlab and couldnt seem to get it going but the template worked a charm. 

I'm glad to see you passionate about the project and hope to be of some sort of contribution one way or another. Been waiting for an awesome working version such as this. Over the years theres been a lot of interesting similar type builds and apps to go along with them but I have to say this seems to be the most promising and proper way to go about it I've seen, Much prefer this over having to have addons running over kodi personally. 

If there is a donate button lead me to it!
